tools/ResourceUpdater/main.cpp

```diff
@@ -763,17 +773,32 @@ bool generate_english_roguelike_stage_name_replacement(const std::filesystem::pa
     return true;
 }
 
-bool update_battle_chars_info(const std::filesystem::path& input_dir, const std::filesystem::path& output_dir)
+bool update_battle_chars_info(const std::filesystem::path& input_dir, const std::filesystem::path& overseas_dir, const std::filesystem::path& output_dir)
 {
-    const auto& input_chars_file = input_dir / "gamedata" / "excel" / "character_table.json";
-    const auto& input_range_file = input_dir / "gamedata" / "excel" / "range_table.json";
+    auto chars_cn_opt = json::open(input_dir / "gamedata" / "excel" / "character_table.json");
+    auto chars_en_opt = json::open(overseas_dir / "en" / "gamedata" / "excel" / "character_table.json");
+    auto chars_jp_opt = json::open(overseas_dir / "jp" / "gamedata" / "excel" / "character_table.json");
+    auto chars_kr_opt = json::open(overseas_dir / "kr" / "gamedata" / "excel" / "character_table.json");
+    auto chars_tw_opt = json::open(overseas_dir / "tw" / "gamedata" / "excel" / "character_table.json");
+
+    auto range_opt = json::open(input_dir / "gamedata" / "excel" / "range_table.json");
 
-    auto chars_opt = json::open(input_chars_file);
-    auto range_opt = json::open(input_range_file);
-    if (!chars_opt || !range_opt) {
+    if (!chars_cn_opt || !chars_en_opt || !chars_jp_opt || !chars_kr_opt || !chars_tw_opt || !range_opt) {
         return false;
     }
-    auto& chars_json = chars_opt.value();
+
+    // auto& chars_cn_json = chars_cn_opt.value();
+    // auto& chars_en_json = chars_en_opt.value();
+    // auto& chars_jp_json = chars_jp_opt.value();
+    // auto& chars_kr_json = chars_kr_opt.value();
+    // auto& chars_tw_json = chars_tw_opt.value();
+
+    std::vector<std::pair<json::value, std::string>> chars_json = { { chars_cn_opt.value(), "name" },
+                                                                    { chars_en_opt.value(), "name_en" },
+                                                                    { chars_jp_opt.value(), "name_jp" },
+                                                                    { chars_kr_opt.value(), "name_kr" },
+                                                                    { chars_tw_opt.value(), "name_tw" } };
+
     auto& range_json = range_opt.value();
```

Comment:
This is probably going overkill. This is scalable, but I don't think new clients are ever gonna get released
```cpp
    if (!range_opt) {
        return false;
    }

    std::vector global = { "en", "jp", "kr", "tw" };
    std::vector chars_opt = { json::open(input_dir / "gamedata" / "excel" / "character_table.json") };

    for (int i = 1; i < 5; i++) {
        chars_opt.push_back(json::open(overseas_dir / global[i - 1] / "gamedata" / "excel" / "character_table.json"));
    }

    for (auto& server : chars_opt) {
        if (!server) {
            return false;
        }
    }

    std::vector<std::pair<json::value, std::string>> chars_json;
    std::vector name_vector = { "name", "name_en", "name_jp", "name_kr", "name_tw" };

    for (int i = 0; i < chars_opt.size(); i++) {
        chars_json.push_back({ chars_opt[i].value(), name_vector[i] });
    }
```